### PR TITLE
docs: format type options

### PR DESCRIPTION
**Pull request by soyeric128**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

- Created a separate page to describe FILE_FORMAT including (Type + formatTypeOptions)
- Removed duplicate content from COPY INTO and Streaming Load API.
- Created reference links to the new page from the COPY INTO and Streaming Load API pages.

Closes #issue

### CODE REVIEW

**Comment by BohuTANG** [comment]
image


The `Input File Formats` is in integration, maybe not correct.

File: docs/doc/14-sql-commands/10-dml/dml-copy-into-table.md [hunk 1]
```diff
@@ -153,146 +153,9 @@ Specifies a list of one or more files names (separated by commas) to be loaded.
 
 A [PCRE2](https://www.pcre.org/current/doc/html/)-based regular expression pattern string, enclosed in single quotes, specifying the file names to match. Click [here](#loading-data-with-pattern-matching) to see an example. For PCRE2 syntax, see http://www.pcre.org/current/doc/html/pcre2syntax.html. 
 
-### formatTypeOptions
+### FILE_FORMAT
```
Comment (BohuTANG) [inline docs/doc/14-sql-commands/10-dml/dml-copy-into-table.md:156]: We should keep these content in the `docs/doc/14-sql-commands/10-dml/dml-copy-into-table.md`, why move it out?

Comment (BohuTANG) [reply docs/doc/14-sql-commands/10-dml/dml-copy-into-table.md:156]: snow example: https://docs.snowflake.com/en/sql-reference/sql/copy-into-table.html#format-type-options-formattypeoptions

File: docs/doc/11-integrations/30-file-format-options.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
+---
+title: Input File Formats
```
Comment (youngsofun) [inline docs/doc/11-integrations/30-file-format-options.md:2]: most options works for both input/format.
Some only for input/output should be pointed output alone.

File: docs/doc/11-integrations/30-file-format-options.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+```
+
+## CSV Options
+
+Databend accepts CVS files that are compliant with [RFC 4180](https://www.rfc-editor.org/rfc/rfc4180) and is subject to the following conditions:
+
+- A string must be quoted if it contains the character of a [QUOTE](#quote), [ESCAPE](#escape), [RECORD_DELIMITER](#record_delimiter), or [FIELD_DELIMITER](#field_delimiter).
+- No character will be escaped in a quoted string except `Quote`.
+- No space should be left between a `FIELD_DELIMITER` and a `Quote`.
+- A record should not contain a trailing `FIELD_DELIMITER`.
```
Comment (youngsofun) [inline docs/doc/11-integrations/30-file-format-options.md:34]:
```suggestion
```

Comment (youngsofun) [reply docs/doc/11-integrations/30-file-format-options.md:34]: trailing `FIELD_DELIMITER` is accepted after some discuss.

File: docs/doc/11-integrations/30-file-format-options.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+  - `ENCLOSED BY` is not empty.
+:::note
+Files will be recognized as the TSV format if the conditions above are not satisfied. For more information about the clauses `ESCAPED BY` and `ENCLOSED BY`, refer to https://dev.mysql.com/doc/refman/8.0/en/load-data.html.
+:::
+
+### RECORD_DELIMITER
+
+Separates records in an input file.
+
+**Available Values**: `\r`,`\n`, or use a character with the escape char: `\b`, `\f`, `\r`, `\n`, `\t`, `\0`, `\xHH`
```
Comment (youngsofun) [inline docs/doc/11-integrations/30-file-format-options.md:48]: any visible character like `a` or `|` is also ok.

Comment (youngsofun) [reply docs/doc/11-integrations/30-file-format-options.md:48]: the limitation is  "one character" except `\r\n` 
`\b`, `\f`, `\r`, `\n`, `\t`, `\0`, `\xHH` are acceptable escaped form of  "one character"

File: docs/doc/11-integrations/30-file-format-options.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+
+**Default**: `\n`
+
+### FIELD_DELIMITER
+
+Separates fields in an input file.
```
Comment (youngsofun) [inline docs/doc/11-integrations/30-file-format-options.md:54]:
```suggestion
Separates fields in record.
```

File: docs/doc/14-sql-commands/10-dml/dml-copy-into-table.md (hunk 1 above)
Comment (soyeric128) [reply docs/doc/14-sql-commands/10-dml/dml-copy-into-table.md:156]: The reason to move it out is FILE_FORMAT is now a common parameter and is shared by the streaming load API as well. 
https://github.com/datafuselabs/databend/pull/9255#issuecomment-1352541314

File: docs/doc/11-integrations/30-file-format-options.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+
+**Available Values**: Use a character with the escape char: `\b`, `\f`, `\r`, `\n`, `\t`, `\0`, `\xHH`
```
Comment (youngsofun) [inline docs/doc/11-integrations/30-file-format-options.md:56]: ditto

File: docs/doc/11-integrations/30-file-format-options.md [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+
+Separates records in an input file.
+
+**Available Values**: `\r`,`\n`, or use a character with the escape char: `\b`, `\f`, `\r`, `\n`, `\t`, `\0`, `\xHH`
+
+**Default**: `\n`
+
+### FIELD_DELIMITER
+
+Separates fields in an input file.
```
Comment (youngsofun) [inline docs/doc/11-integrations/30-file-format-options.md:127]:
```suggestion
Separates fields in a record.
```

File: docs/doc/11-integrations/30-file-format-options.md [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+- These characters in a TSV file will be escaped: `\b`, `\f`, `\r`, `\n`, `\t`, `\0`, `\\`, `\'`, [RECORD_DELIMITER](#record_delimiter-1), [FIELD_DELIMITER](#field_delimiter-1).
+- Neither quoting nor enclosing is currently supported.
+- A string will be quoted in CSV if it comes from a serialized Array or Struct field.
+- Null is serialized as `\N`.
+
+### RECORD_DELIMITER
+
+Separates records in an input file.
+
+**Available Values**: `\r`,`\n`, or use a character with the escape char: `\b`, `\f`, `\r`, `\n`, `\t`, `\0`, `\xHH`
```
Comment (youngsofun) [inline docs/doc/11-integrations/30-file-format-options.md:121]: ditto

File: docs/doc/11-integrations/30-file-format-options.md (hunk 2 above)
Comment (soyeric128) [reply docs/doc/11-integrations/30-file-format-options.md:2]: Could you tell me which ones are for input or output ONLY?

File: docs/doc/11-integrations/00-api/03-streaming-load.md [hunk 9]
```diff
@@ -18,17 +18,13 @@ curl -H "insert_sql:<value>" -F "upload=@<file_location>" [-F "upload=@<file_loc
 ```
 ## Explaining Argument `-H`
 
-The request usually includes many occurrences of the argument `-H` and each is followed by one of the following parameters to tell Databend how to handle the file you're loading data from. Please note that `insert_sql` is required.
+The request usually includes many occurrences of the argument `-H` and each is followed by one of the following parameters to tell Databend how to handle the file you're loading data from. Please note that `insert_sql` is required. For the parameter `FILE_FORMAT` in `insert_sql`, see [Input File Formats](../30-file-format-options.md).
```
Comment (youngsofun) [inline docs/doc/11-integrations/00-api/03-streaming-load.md:21]: 1. `-H` is a general option of curl to pass HTTP header,  current desc seems like it databend special, and is not Professional. 
2.  we only use  `insert_sql` header
3. FILE_FORMAT cause is used in the SQL

 I  curently we only use `insert_sql` header to pass the insert sql with FILE_FORMAT.
for curl, just used `-H`, like ...
for options of FILE_FORMAT refer to ..
```

File: docs/doc/11-integrations/30-file-format-options.md (hunk 2 above)
Comment (youngsofun) [reply docs/doc/11-integrations/30-file-format-options.md:2]: https://github.com/datafuselabs/databend/blob/main/docs/doc/14-sql-commands/10-dml/dml-copy-into-table.md#skip_header--integer

XML format is now input only

CSV quote and skip_header options is  input only

Comment (youngsofun) [reply docs/doc/11-integrations/30-file-format-options.md:2]: snowflake docs use  **Usage**  for this

File: docs/doc/11-integrations/00-api/03-streaming-load.md (hunk 9 above)
Comment (youngsofun) [reply docs/doc/11-integrations/00-api/03-streaming-load.md:21]: cc @Xuanwo  @BohuTANG

File: docs/doc/11-integrations/30-file-format-options.md (hunk 4 above)
Comment (soyeric128) [reply docs/doc/11-integrations/30-file-format-options.md:47]: **Available Values**: 
- `\r\n`
- An arbitrary character, such as `a` and `|`.
- A character with the escape char: `\b`, `\f`, `\r`, `\n`, `\t`, `\0`, `\xHH`

File: docs/doc/11-integrations/30-file-format-options.md (hunk 2 above)
Comment (soyeric128) [reply docs/doc/11-integrations/30-file-format-options.md:2]: The "output" refers to unloading data from Databend, right? @youngsofun

File: docs/doc/11-integrations/00-api/03-streaming-load.md (hunk 9 above)
Comment (soyeric128) [reply docs/doc/11-integrations/00-api/03-streaming-load.md:21]: I didnt know that -H is part of CURL. I will re-edit it.

File: docs/doc/11-integrations/30-file-format-options.md (hunk 4 above)
Comment (youngsofun) [reply docs/doc/11-integrations/30-file-format-options.md:47]: ok to me.

File: docs/doc/11-integrations/30-file-format-options.md (hunk 2 above)
Comment (youngsofun) [reply docs/doc/11-integrations/30-file-format-options.md:2]: > The "output" refers to unloading data from Databend, right? @youngsofun 

yes

File: docs/doc/14-sql-commands/10-dml/dml-copy-into-table.md (hunk 1 above)
Comment (soyeric128) [reply docs/doc/14-sql-commands/10-dml/dml-copy-into-table.md:156]: Feishu20221221-091917

As discussed with @BohuTANG , the new page is to be moved here.

**Review by BohuTANG — APPROVED** [review]

**Comment by BohuTANG** [comment]
image
Almost every time there is an error after merging to the main branch, we need to deal with it. @PsiACE @Xuanwo

**Pull request merged**